src/main/java/networkbook/model/person/Priority.java

```diff
@@ -0,0 +1,103 @@
+package networkbook.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static networkbook.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's priority level in the network book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPriority(PriorityLevel)}
+ */
+public class Priority {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Priority is among {high, medium, low}, either the word or the first letter, and is not case-sensitive";
+
+    private final PriorityLevel value;
+
+    /**
+     * Constructs a {@code Priority}.
+     *
+     * @param priorityString A valid String representation of a priority level.
+     */
+    public Priority(String priorityString) {
+        requireNonNull(priorityString);
+        PriorityLevel priorityLevel = parsePriorityLevel(priorityString);
+        checkArgument(isValidPriority(priorityLevel), MESSAGE_CONSTRAINTS);
+        value = priorityLevel;
+    }
+
+    /**
+     * Parses user input {@code priorityString} into a {@code PriorityLevel}
+     *
+     * @param priorityString provided in user command
+     * @return corresponding PriorityLevel
+     */
+    public static PriorityLevel parsePriorityLevel(String priorityString) {
+        priorityString = priorityString.toLowerCase();
+        switch (priorityString) {
+        case "high":
+        case "h":
+            return Priority.PriorityLevel.HIGH;
+        case "medium":
+        case "m":
+            return Priority.PriorityLevel.MEDIUM;
+        case "low":
+        case "l":
+            return Priority.PriorityLevel.LOW;
+        default:
+            return Priority.PriorityLevel.INVALID;
+        }
+    }
+
+    /**
+     * Returns true if a given priority level is valid.
+     */
+    public static boolean isValidPriority(PriorityLevel priorityLevel) {
+        return priorityLevel != PriorityLevel.INVALID;
+    }
+
+    @Override
+    public String toString() {
+        switch (value) {
+        case HIGH:
+            return "High";
+        case MEDIUM:
+            return "Medium";
+        case LOW:
+            return "LOW";
+        default:
+            return "Invalid priority level";
```

Comment:
Should use assertions to indicate programming error.

Reply:
Yes, assertion is used to indicate programming error. When a Priority object is instantiated, the static `isValidPriority` method will be called. The use of `PriorityLevel.INVALID` is just to avoid writing a duplicated method to check validity.

This overriding toString() method should not throw an exception. Thus, in the unlikely case that the above validation process is wrong due to programming error, toString() will return "invalid" to notify the programmer.

LMK if you have a different opinion or other suggestions.

Reply:
Yes, it should not throw an exception, but I did not say so? Here it would be useful to add an assert statement to indicate a programming error. In Java syntax:
```java
default:
    assert false: "Priority level should fall under one of the three cases";
```

Reply:
Oh I see, sorry I saw wrongly. You are right that an assertion here would be more noticeable than a string. I'll include this suggestion in my next PR. Thanks!